src/strategy-management/Toolkit.ts

```diff
@@ -935,6 +935,22 @@ export class Toolkit {
     );
     const encStrategy = encodeStrategy(strategy);
 
+    // if exactly one of the encStrategy orders has y value 0  we will set its z value using calculateRequiredLiquidity
+    if (encStrategy.order0.y.isZero() && !encStrategy.order1.y.isZero()) {
+      encStrategy.order0.z = calculateRequiredLiquidity(
+        strategy.order1,
+        strategy.order0
+      );
+    } else if (
+      !encStrategy.order0.y.isZero() &&
+      encStrategy.order1.y.isZero()
+    ) {
+      encStrategy.order1.z = calculateRequiredLiquidity(
+        strategy.order0,
+        strategy.order1
+      );
+    }
```

Comment:
The returned value of function `calculateRequiredLiquidity` should not be assigned to `z`.
It should be used as the missing `liquidity` before passing that order to function `encodeOrder`.
The value of `z` will then be calculated internally in that function.
And of course, the user will need to provide that "missing liquidity".

If we're looking to keep `y` at 0 but still set `z` to the desirable value, then there are two options:
1. In function `encodeOrder`, pass an additional flag indicating that the function should set `y` to 0, but still use the input `liquidity` for the calculation of `z`
2. In function `calculateRequiredLiquidity`, return not only the missing liquidity, but also an intermediate value that this function calculates, which holds precisely the `z` value that the future encoded order will ultimately hold

Reply:
Well, for 2, we might even get rid of the current return valuer of that function, if we determine that it is not needed anywhere (based on product requirements, etc).